src/CoreWCF.RabbitMQ.Client/src/CoreWCF/ServiceModel/Channels/RabbitMqBinding.cs

```diff
@@ -0,0 +1,113 @@
+using Binding = System.ServiceModel.Channels.Binding;
+using BindingElementCollection = System.ServiceModel.Channels.BindingElementCollection;
+using TextMessageEncodingBindingElement = System.ServiceModel.Channels.TextMessageEncodingBindingElement;
+
+namespace CoreWCF.ServiceModel.Channels
+{
+    public class RabbitMqBinding : Binding
+    {
+        private RabbitMqTransportBindingElement _transport;
+        private TextMessageEncodingBindingElement _textMessageEncodingBindingElement;
+        private BinaryMessageEncodingBindingElement _binaryMessageEncodingBindingElement;
+
+        public RabbitMqBinding(string uri) : this(new Uri(uri))
+        { }
+
+        public RabbitMqBinding(Uri uri)
+        {
+            _textMessageEncodingBindingElement = new TextMessageEncodingBindingElement();
+            _binaryMessageEncodingBindingElement = new BinaryMessageEncodingBindingElement();
+
+            var sslOption = new SslOption();
+            var virtualHost = ConnectionFactory.DefaultVHost;
+            var credentials = new NetworkCredential(ConnectionFactory.DefaultUser, ConnectionFactory.DefaultPass);
+
+            _transport = new RabbitMqTransportBindingElement
+            {
+                BaseAddress = uri,
+                Credentials = credentials,
+                SslOption = sslOption,
+                VirtualHost = virtualHost
+            };
+        }
+
+        /// <summary>
+        /// Specifies the maximum encoded message size
+        /// </summary>
+        public long MaxMessageSize
+        {
+            get => _transport.MaxReceivedMessageSize;
+            set => _transport.MaxReceivedMessageSize = value;
+        }
+
+        /// <summary>
+        /// Specifies the version of the AMQP protocol that should be used to communicate with the broker
+        /// </summary>
+        public IProtocol BrokerProtocol
+        {
+            get => _transport.BrokerProtocol;
+            set => _transport.BrokerProtocol = value;
+        }
+
+        public Uri BaseAddress
+        {
+            get => _transport.BaseAddress;
+            set => _transport.BaseAddress = value;
+        }
+
+        public SslOption SslOption
+        {
+            get => _transport.SslOption;
+            set => _transport.SslOption = value;
+        }
+
+        public string VirtualHost
+        {
+            get => _transport.VirtualHost;
+            set => _transport.VirtualHost = value;
+        }
+
+        /// <summary>
+        /// Credentials to access the RabbitMQ host
+        /// </summary>
+        public ICredentials Credentials
+        {
+            get => _transport.Credentials;
+            set => _transport.Credentials = value;
+        }
+
+        /// <summary>
+        /// Gets the scheme used by the binding
+        /// </summary>
+        public override string Scheme => _transport.Scheme;
+
+        public RabbitMqMessageEncoding MessageEncoding { get; set; } = RabbitMqMessageEncoding.Text;
+
```

Comment:
Need to validate on the setter that a valid value is being set. Any int can be cast to an enum. I could do `rabbitMqBinding.MessageEncoding = (RabbitMqMessageEncoding)27;` and the value would be set to something which doesn't make sense. Take a look at the code [here](https://source.dot.net/#System.ServiceModel.Http/System/ServiceModel/BasicHttpMessageSecurity.cs,25) to see a typical pattern.